From e7ae44999fcd7c3006233ece57f145ed635b9fd3 Mon Sep 17 00:00:00 2001
From: zhang xingxing <2388969553@qq.com>
Date: Sat, 10 Oct 2026 08:31:03 +0200
Subject: [PATCH] avformat/iamf_reader: fix the inverted subblock duration
 check

This is not a security issue.

Regression since: 8665748584
Fixes: usWfwH9F2jUN
Fixes: swarm-Inverted-comparison-in-subblock_duration-validation-6190ea
Found-by: zhang xingxing <2388969553@qq.com>
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
(cherry picked from commit e7ae44999fcd7c3006233ece57f145ed635b9fd3)
---
 libavformat/iamf_reader.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/iamf_reader.c b/libavformat/iamf_reader.c
index 1b963ee311b..c2375ea8356 100644
--- a/libavformat/iamf_reader.c
+++ b/libavformat/iamf_reader.c
@@ -201,7 +201,7 @@ static int parameter_block_obu(AVFormatContext *s, IAMFDemuxContext *c,
 
         if (!param_definition->mode && !constant_subblock_duration) {
             subblock_duration = ffio_read_leb(pb);
-            if (duration - total_duration > subblock_duration) {
+            if (subblock_duration > duration - total_duration) {
                 av_log(s, AV_LOG_ERROR, "Invalid subblock durations in parameter_id %u\n", parameter_id);
                 ret = AVERROR_INVALIDDATA;
                 goto fail;
-- 
2.53.0

